plasma_cash/utils/merkle/sparse_merkle_tree.py

```diff
@@ -0,0 +1,37 @@
+from collections import OrderedDict
+from ethereum.utils import sha3
+
+
+class SparseMerkleTree(object):
+
+    def __init__(self, leaves={}):
+        self.depth = 256
+        self.leaves = OrderedDict(sorted(leaves.items(), key=lambda t: t[0]))
+        self.default_nodes = [b'\x00' * 32]
+        self.create_default_nodes()
+        if bool(leaves):
+            self.root = self.create_tree()
+        else:
+            self.root = self.default_nodes[self.depth - 1]
+
+    def create_default_nodes(self):
+        for i in range(1, self.depth):
+            prev_default = self.default_nodes[i - 1]
+            self.default_nodes.append(sha3(prev_default + prev_default))
+
+    def create_tree(self):
+        tree_level = self.leaves
+        for i in range(self.depth - 1):
+            next_level = {}
+            prev_index = -1
+            for index, value in tree_level.items():
+                if index % 2 == 0:
```

Comment:
Although I am not a fan of using comment, would be nice here to tell what are these if else relationship in tree structure IMHO

Reply:
You're right. I'll add some comments.